### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by mzbroch**
Documentation updates:
- FAQ update
- Onboarding Extensions update

### CODE REVIEW

**Review by dgarros — APPROVED** [review]
love it

File: FAQ.md [hunk 1]
```diff
@@ -20,25 +29,32 @@
...
+
+## How can I update the optional arguments for NAPALM ?
+
+> Optional arguments are often used to define a `secret` for Cisco devices and other connection parameters. By default, plugin will use a provided secret for each onboarding task. If such one is not provided, for tasks with a declared platform plugin will read optional arguments from Nautobot if they are defined at a platform level. Last resort of optional arguments is `settings.NAPALM_ARGS`
 
 ## Does this plugin support the discovery and the creation of all interfaces and IP Addresses ?
 
 > **No**, The plugin will only discover and create the management interface and the management IP address. Importing all interfaces and IP addresses is a much larger problem that requires more preparation. This is out of scope of this project.
 
+> We recommend Network Importer tool from Network to Code for a post-onboarding network state synchronization. See [https://github.com/networktocode/network-importer](url) for more details
```
Comment (glennmatthews) [inline FAQ.md:42]:
```suggestion
> We recommend Network Importer tool from Network to Code for a post-onboarding network state synchronization. See [its GitHub repository](https://github.com/networktocode/network-importer) for more details
```

File: FAQ.md [hunk 2]
```diff
@@ -20,25 +29,32 @@
...
+
+
 ## Does this plugin support the discovery of device based on fqdn ? 
 
-> **No**, Current the onbarding process is based on an IP address, please open an issue to discuss your use case if you would like to see support for FQDN based devices too. 
+> **Yes**, plugin will resolve FQDN into an IP-address and will use the IP-address for its connections.
```
Comment (glennmatthews) [inline FAQ.md:47]:
```suggestion
> **Yes**, plugin will resolve FQDN into an IP address and will use the IP address for its connections.
```

File: docs/onboarding-extensions/onboarding_extensions.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- Stack onboarding
+- HA Pairs, VSS pairs onboarding
+- Populating device-bays and child devices, ie. Cisco UCS-Es
+- Populating device inventories during onboarding
+- Setting device roles during onboarding, ie. `Role = Access Switch`
+- Setting device tags during onboarding, ie: `onboarded_with=networktocode`
+
+## How do I enable onboarding extensions ?
+
+> You have to modify Nautobot configuration in `nautobot_config.py` to include yours extensions configuration. In the example below, we map a NAPALM driver name (this one could be a custom NAPALM driver too!), into a python loadable module. Onboarding plugin will dynamically load and execute your module as specified below:
```
Comment (glennmatthews) [inline docs/onboarding-extensions/onboarding_extensions.md:19]:
```suggestion
> You have to modify Nautobot configuration in `nautobot_config.py` to include your extension's configuration. In the example below, we map a NAPALM driver name (this one could be a custom NAPALM driver too!), into a python loadable module. Onboarding plugin will dynamically load and execute your module as specified below:
```

File: docs/onboarding-extensions/onboarding_extensions.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+```
+        "onboarding_extensions_map": {
+            "your_napalm_driver_name_ios": "onboarding_extensions.platforms.your_napalm_driver_name_ios",
+            "your_napalm_driver_name_nxos_ssh": "onboarding_extensions.platforms.your_napalm_driver_name_nxos_ssh",
+            "your_napalm_driver_name_asa_ssh": "onboarding_extensions.platforms.your_napalm_driver_name_asa_ssh",
```
Comment (glennmatthews) [inline docs/onboarding-extensions/onboarding_extensions.md:24]: This example could be clearer, IMHO. Is the idea that `"your_napalm_driver_name_ios"` is the platform's configured NAPALM driver name? Or if not, what does this string correspond to?

File: docs/onboarding-extensions/onboarding_extensions.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        },
+```
+
+## How do I create an onboarding extension ?
+
+> You have to create two elements as a part of your extension: an `OnboardingDriverExtensions` class inside your python module, and the `OnboardingClass` itself.
+
+> At first, plugin will initiate an instance of your `OnboardingDriverExtensions` (you have to use this name of the class). As we pass the NAPALM device with an opened connection to this class, you can leverage NAPALM to execute any custom commands (ie. collect device stack information). The are two important class attributes: `onboarding_class` (mandatory) and `ext_result` (optional). In the `onboarding_class`, you have to return a custom class that defines your onboarding methods.
```
Comment (glennmatthews) [inline docs/onboarding-extensions/onboarding_extensions.md:32]: Perhaps explain what purpose `ext_result` serves and how it is to be used?

**Comment by mzbroch** [comment]
Thank You @glennmatthews - all changes & suggestions applied !

**Pull request merged**